src/json-schema/partial/map_state.json

```diff
@@ -49,6 +49,219 @@
+                            "properties": {
+                                "ReaderConfig": {
+                                    "type": "object",
+                                    "description": "A JSON object that specifies where to read the items instead of from the effective input.",
+                                    "properties": {
+                                        "InputType": {
+                                            "type": "string",
+                                            "description": "A string that specifies the type of Amazon S3 data source.",
+                                            "minLength": 1,
+                                            "examples": [
+                                                "CSV",
+                                                "JSON"
+                                            ]
+                                        },
+                                        "CSVHeaders": {
+                                            "type": "array",
+                                            "minItems": 1,
+                                            "items": {
+                                                "type": "string"
+                                            }
+                                        },
+                                        "CSVHeaderLocation": {
+                                            "type": "string",
+                                            "description": "A string that specifies the location of the CSV column header.",
+                                            "minLength": 1,
+                                            "enum": [
+                                                "FIRST_ROW",
+                                                "GIVEN"
+                                            ]
+                                        },
+                                        "MaxItems": {
+                                            "type": "integer",
+                                            "description": "An integer that limits the number of data items passed to the Map state.",
+                                            "minimum": 0
+                                        },
+                                        "MaxItemsPath": {
+                                            "$ref": "common.json#/definitions/referencePath",
+                                            "description": "A reference path to an integer that limits the number of data items passed to the Map state."
+                                        }
+                                    },
+                                    "required": [
+                                        "InputType"
+                                    ]
+                                },
+                                "Resource": {
+                                    "type": "string",
```

Comment:
Can we use an enum for this as well? I believe only `GetObject` and `ListObjectsV2` are supported.

Reply:
Makes sense, let's leave these as is.